src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/SeedFromExcelController.kt

```diff
@@ -0,0 +1,19 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.controller
+
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.SeedFromExcelApiDelegate
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.api.model.SeedFromExcelRequest
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.seed.SeedService
+
+@Service
+class SeedFromExcelController(private val seedService: SeedService) : SeedFromExcelApiDelegate {
+  override fun seedFromExcelPost(seedFromExcelRequest: SeedFromExcelRequest): ResponseEntity<Unit> {
+    throwIfNotLoopbackRequest()
+
+    seedService.seedExcelData(seedFromExcelRequest.seedType, seedFromExcelRequest.premisesId, seedFromExcelRequest.fileName)
```

Comment:
I was thinking to pass the premise id as an argument to the run seed script to begin with (haven't done this yet), seemed the most straightforward thing to do.

Then run the script for every spreadsheet rather than for every file in a directory?

Reply:
Probably one to return to later. We have 108 premises, most of which (presumably) we'd want to load in bulk. We could always just script a curl command to be fair!

Reply:
Ticket [APS-1733](https://dsdmoj.atlassian.net/browse/APS-1733) raised to investigate simplifying the existing script.